### DISCUSSION TITLE
Octoprint - Specific USB pass trough and webcams

### DISCUSSION THREAD

**Discussion by DocN0** [tteck/Proxmox#1942] [Q&A]
Hi all  
I have two questions regarding the Octoprint installation. 

Is it possible to pass only one specific USB port to the LXC? I would like to run two Ocotoprint instances for two printers. 
Now both Ocotoprint instances find both printers which makes it a bit tricky. 

What about USB webcams? no common stream URL seems to work, is there anything else to consider? 

Thanks a lot

**Comment by tteck** [comment]
I don't use Octoprint, so this may not work, but Octoprint has a place to define "Additional serial ports" in settings. Can you define the device by-id?
If you execute this `ls -l /dev/serial/by-id` in the LXC console does it list the devices? 
I can execute the command and it shows the only USB device that I have (Zigbee stick) 
```
root@octoprint:~# ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 Oct 13 11:48 usb-Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_2e893de28e93eb11842d214f3d9-if00-port0 -> ../../ttyUSB0
```
So, I'd add `/dev/serial/by-id/Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_2e893de28e93eb11842d214f3d9-if00-port0` in the section "Additional serial ports"  Then it would be accessable from the "Serial Port" dropdown.

Just a thought.

**Reply by DocN0** [reply]
thank you, currently i only have one printer attached but i want to change that and replace the other pi. 

currently i have a printer attached to this proxmox node as well as a USB webcam 
however my output from "ls -l /dev/serial/by-id" is only: 

```
root@octoprint:~# ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 Oct 21 22:26 usb-Silicon_Labs_CP2102_USB_to_UART_Bridge_Controller_0001-if00-port0 -> ../../ttyUSB0
root@octoprint:~# 
```

it looks like not even the usb cam is recognized, printing works fine so far 
only what is currently still a problem, is when the 3d printer is switched off and the LXC is still running no connection to the printer can be established without restarting the LXC. 

in octoprint there is the possibility to define shutdown and restart commands but these do not start the LXC in my case.

**Comment by DocN0** [comment]
@tteck do i need to tag you to get a notification?
 still relatively new to github, anyway thanks for your reply. :)

**Comment by hugo18192** [comment]
Hello,

what about webcams? Can these also be passed on?
Thanks a lot
